canvasapi/course.py

```diff
@@ -286,24 +284,21 @@ def create_external_tool(
 
         :param name: The name of the tool
         :type name: str
-        :param privacy_level: What information to send to the external
-            tool. Options are "anonymous", "name_only", "public"
-        :type privacy_level: str
-        :param consumer_key: The consumer key for the external tool
-        :type consumer_key: str
-        :param shared_secret: The shared secret with the external tool
-        :type shared_secret: str
+
         :rtype: :class:`canvasapi.external_tool.ExternalTool`
         """
         from canvasapi.external_tool import ExternalTool
 
+        required_params = ("name", "privacy_level", "consumer_key", "shared_secret")
```

Comment:
I wonder if we could write a utility to wrap our `RequiredFieldMissing` code. Definitely outside the scope of this patch, but it seems like I've seen this pattern before. Food for thought.

Reply:
Not a bad idea. We handle it slightly differently in a lot of places, might be useful to standardize.